### PR TITLE
Update Ruby client

### PR DESCRIPTION
**Pull request by wolf31o2**
Lots of updates
- Rename gem to `cdap-stream-client` from `stream-client-ruby`
- Rename module to `CDAP` from `CDAPIngest`
- Rename `Rest` class to `RestClient`
- Let `rubocop` autocorrect a bunch of things
- Update License file

Otherwise, there's no other code updates or changes.

### CODE REVIEW

File: cdap-stream-clients/ruby/LICENSE.txt [hunk 1]
```diff
@@ -0,0 +1,232 @@
+Copyright © 2014-2015 Cask Data, Inc.
```
Comment (awholegunch) [inline cdap-stream-clients/ruby/LICENSE.txt]: 2016

File: cdap-stream-clients/ruby/cdap-stream-client.gemspec [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+$LOAD_PATH.push File.expand_path('../lib', __FILE__)
+require 'cdap-stream-client/version'
+
+Gem::Specification.new do |s|
+  s.name        = 'cdap-stream-client'
+  s.version     = CDAP::VERSION
+  s.authors     = ['Cask Data, Inc.']
+  s.email       = ['ops@cask.co']
+  s.summary     = 'A Ruby client for interacting with Cask CDAP streams'
+  s.description = 'Provide a more Ruby experience when using streams in Cask CDAP services.'
```
Comment (awholegunch) [inline cdap-stream-clients/ruby/cdap-stream-client.gemspec]: Cask CDAP -> CDAP

Comment (awholegunch) [reply cdap-stream-clients/ruby/cdap-stream-client.gemspec]: Two places 25 & 26

**Comment by awholegunch** [comment]
Minor comments, LGTM.

File: cdap-stream-clients/ruby/LICENSE.txt (hunk 1 above)
Comment (wolf31o2) [reply cdap-stream-clients/ruby/LICENSE.txt]: Copied that from CDAP, but sure.

**Pull request merged**